platform/kube/conversion.go

```diff
@@ -124,3 +128,104 @@ func convertProtocol(name string, proto v1.Protocol) model.Protocol {
 	}
 	return out
 }
+
+func convertProbePort(c v1.Container, port intstr.IntOrString) (int, error) {
+	switch port.Type {
+	case intstr.Int:
+		return port.IntValue(), nil
+	case intstr.String:
+		for _, named := range c.Ports {
+			if named.Name == port.String() {
+				return int(named.ContainerPort), nil
+			}
+		}
+		return 0, fmt.Errorf("missing named port %q", port)
+	default:
+		return 0, fmt.Errorf("incorrect port type %q", port)
+	}
+}
+
+// convertProbesToPorts returns a PortList consisting of the ports where the
+// pod is configured to do Liveness and Readiness probes
+func convertProbesToPorts(t *v1.PodSpec) (model.PortList, error) {
+	set := make(map[string]*model.Port)
+	var errs error
+	for _, container := range t.Containers {
+		if container.LivenessProbe != nil {
+			if container.LivenessProbe.HTTPGet != nil {
+				port, err := convertProbePort(container, container.LivenessProbe.Handler.HTTPGet.Port)
+				if err != nil {
+					errs = multierror.Append(errs, err)
+				} else {
+					p := &model.Port{
+						Name:     "mgmt-" + strconv.Itoa(port),
+						Port:     port,
+						Protocol: model.ProtocolHTTP,
+					}
+					// Deduplicate along the way. We don't differentiate between HTTP vs TCP mgmt ports
+					if set[p.Name] == nil {
+						set[p.Name] = p
+					}
+				}
+			} else if container.LivenessProbe.TCPSocket != nil {
+				// Only one type of handler is allowed by Kubernetes (HTTPGet or TCPSocket)
+				port, err := convertProbePort(container, container.LivenessProbe.TCPSocket.Port)
+				if err != nil {
+					errs = multierror.Append(errs, err)
+				} else {
+					p := &model.Port{
+						Name:     "mgmt-" + strconv.Itoa(port),
+						Port:     port,
+						Protocol: model.ProtocolTCP,
+					}
+					// Deduplicate along the way. We don't differentiate between HTTP vs TCP mgmt ports
+					if set[p.Name] == nil {
+						set[p.Name] = p
+					}
+				}
+			}
+		}
+
+		if container.ReadinessProbe != nil {
```

Comment:
please de-duplicate the code (HTTPGetAction in Probe is same for liveness and readiness)